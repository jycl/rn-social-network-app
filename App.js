import React, { Component } from "react";
import { createStackNavigator, HeaderBackButton } from "react-navigation";
import UserListScreen from "./src/screens/User/ListScreen";
import UserDetailScreen from "./src/screens/User/DetailScreen";
import HomeScreen from "./src/screens/HomeScreen";
import PostCommentScreen from "./src/screens/Post/CommentScreen";
import PhotoGridScreen from "./src/screens/Photo/GridScreen";
import PhotoDetailScreen from "./src/screens/Photo/DetailScreen";
import { Provider, observer } from "mobx-react";
import stores from "./src/stores";
import colors from "./src/styles/colors";
import fontSize from "./src/styles/fontSize";
import userStore from "./src/stores/userStore";

//callback to update the navBar header title value
const navStyleRenderer = () => ({
  title: `${userStore.selectedUser.name}`
});
const AppNavigator = createStackNavigator(
  {
    Home: { screen: HomeScreen },
    UserList: {
      screen: UserListScreen,
      navigationOptions: ({ navigation }) => ({
        title: `User List`,
        headerTitleStyle: {
          fontSize: fontSize.large
        }
      })
    },
    UserDetail: {
      screen: UserDetailScreen,
      navigationOptions: ({ navigation }) => ({
        title: `${userStore.selectedUser.name}`,
        headerLeft: () => {
          let goBack = () => {
            userStore.resetSelection();
            navigation.goBack();
          };
          return (
            <HeaderBackButton
              tintColor={colors.lightestBlue}
              onPress={() => goBack()}
            />
          );
        }
      })
    },
    PostComment: {
      screen: PostCommentScreen,
      navigationOptions: navStyleRenderer
    },
    PhotoGrid: {
      screen: PhotoGridScreen,
      navigationOptions: navStyleRenderer
    },
    PhotoDetail: {
      screen: PhotoDetailScreen,
      navigationOptions: navStyleRenderer
    }
  },
  {
    initialRouteName: "UserList",
    navigationOptions: {
      headerStyle: {
        backgroundColor: colors.darkestBlue
      },
      headerTintColor: colors.lightestBlue,
      headerTitleStyle: {
        fontSize: fontSize.medium
      },
      headerBackTitle: " "
    }
  }
);

@observer
export default class App extends Component {
  render() {
    return (
      <Provider {...stores}>
        <AppNavigator />
      </Provider>
    );
  }
}
