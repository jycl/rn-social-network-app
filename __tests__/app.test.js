import React from "react";
import ReactTestRenderer from "react-test-renderer";
import App from "../App";

describe("App", () => {
  it("renders correctly", () => {
    const instance = ReactTestRenderer.create(<App />);
    expect(instance.toJSON()).toMatchSnapshot();
  });
});
