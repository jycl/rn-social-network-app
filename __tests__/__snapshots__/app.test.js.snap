// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    onMoveShouldSetResponder={[Function]}
    onMoveShouldSetResponderCapture={[Function]}
    onResponderEnd={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderReject={[Function]}
    onResponderRelease={[Function]}
    onResponderStart={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    onStartShouldSetResponderCapture={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column-reverse",
        },
        Object {
          "backgroundColor": "#000",
        },
      ]
    }
  >
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="auto"
        style={
          Object {
            "backgroundColor": "#E9E9EF",
            "bottom": 0,
            "left": 0,
            "marginTop": 0,
            "opacity": 1,
            "position": "absolute",
            "right": 0,
            "shadowColor": "black",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0.2,
            "shadowRadius": 5,
            "top": 0,
            "transform": Array [
              Object {
                "translateX": 0,
              },
              Object {
                "translateY": 0,
              },
            ],
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#F5FCFF",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "blue",
                "borderRadius": 10,
                "opacity": 1,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#FFFFFF",
                  "fontSize": 20,
                  "margin": 10,
                  "textAlign": "center",
                }
              }
            >
              Go to user list
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      pointerEvents="box-none"
    >
      <View
        style={
          Object {
            "backgroundColor": "#F7F7F7",
            "transform": Array [
              Object {
                "translateX": 0,
              },
            ],
          }
        }
      >
        <View
          onLayout={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "backgroundColor": "#F7F7F7",
              "borderBottomColor": "#A7A7AA",
              "borderBottomWidth": 0.5,
              "height": 64,
              "paddingBottom": 0,
              "paddingLeft": 0,
              "paddingRight": 0,
              "paddingTop": 20,
            }
          }
        >
          <View
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
          <View
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "bottom": 0,
                  "flexDirection": "row",
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
            >
              <View
                pointerEvents="box-none"
                style={
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "transparent",
                    "bottom": 0,
                    "flexDirection": "row",
                    "justifyContent": "center",
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  }
                }
              >
                <Text
                  accessibilityTraits="header"
                  allowFontScaling={true}
                  numberOfLines={1}
                  onLayout={[Function]}
                  style={
                    Object {
                      "color": "rgba(0, 0, 0, .9)",
                      "fontSize": 17,
                      "fontWeight": "700",
                      "marginHorizontal": 16,
                      "textAlign": "center",
                    }
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
